[Onepane] Adds Onepane Integration (ECOINT-3) (#2426)
DataDog/integrations-extras: onepane/README.md

## Setup

In Datadog:

- Generate an Application Key:

    - Access your personal settings in Datadog.

    - Navigate to the Application keys section under Organization settings.

    - Select "New Key" to generate a new Application key.

    - Name the key and create it.

    - Copy the generated Application key for future use.

In Onepane:

Here's how to connect your Datadog account to Onepane 

- Access Onepane Console: Head over to the Onepane console at [console.onepane.ai][2]

- Locate Integrations: In the left-hand menu, find the "Integrations" section. Click the "Add Integration" button in the top right corner.

- Choose Datadog: From the list of available integrations, select "Datadog".

- When you select Datadog, you will be redirected to the Datadog login page. 

- Authenticate with OAuth.

    - Log in using your Datadog credentials.

    - Upon successful authentication, Datadog will create an API key for you.

- Provide a name for your Datadog connector.

- Input the site URL and Application key obtained from the ![prerequisite steps][8].

> 

Reviewer: OAuth must be intiated from Datadog. To make OAuth work for users across all Datadog regions, you will need to ensure that you’re making the correct API calls based on their region. **When the user kicks off authorization from the Datadog tile, a site parameter is sent on redirect from the onboarding_url. You will use this site parameter in your calls to the Authorize and Token endpoints. Please note that if a user kicks off authorization directly from your platform, this site parameter will not be sent, and the user will be prompted to select their site on the Datadog authorize page.**

- Finalize Integration: Once satisfied, click "Create" to establish the Datadog connector. It will be deployed shortly.

    ![Deploying][5]

- View Host Resources: After successful deployment, you'll be able to see a list of host resources from your Datadog account within Onepane.

    ![Host Resource][6]

- Map Resources: Onepane allows you to map these Datadog resources to your existing Onepane resources for unified view of incidents and events across your infrastructure

    ![Onepane Mapping][7]

With these steps complete, you'll have successfully integrated Datadog with Onepane

For more details, refer to the [Onepane documentation][2]

## Uninstallation

Here's how to delete the Datadog integration from your Onepane console:

- Access Integrations: Navigate to the "Integrations" section within the Onepane console.

- Locate Datadog: Find the Datadog integration from the list.

- Initiate Deletion: Click the three dots menu (ellipsis) in the top right corner of the Datadog integration tile. Select "Delete" from the options.

- Confirm Deletion: A confirmation dialog might appear. If you're certain you want to proceed, click "Confirm" to remove the Datadog integration.

> When deleting the Datadog integration, the OAuth-generated API keys and other credentials will be soft-deleted and permanently deleted after 7 days.

Reviewer: Please update this to reflect Datadogs OAuth uninstallation steps

# Onepane

## Overview

Onepane is a powerful addition to your existing Datadog, Cloud, and DevOps tools, providing seamless out-of-the-box integrations. It adopts an agentless approach and features automatic asset discovery, allowing you to get value in a matter of hours, not days or weeks. Onepane comprehends your unique landscape, generating a dynamic service map of your operational environment by harnessing data from various systems. This map assists your teams in swiftly resolving incidents and streamlines the audit process. Opt for Onepane to leverage additional ROI on your existing investments. 

- Show changes and infrastructure events alongside errors to pinpoint potential root causes.
- Enable root cause investigation by correlating incidents with change data and infrastructure data.

> ## Setup

Reviewer: OAuth must be kicked off from Datadog. Please include the exact steps a user will take within Datadog to set up the integration. For example, go to the Onepane integration on the integrations page, select install, click connect accounts, log in to your onepane account, .... etc

Reviewer: The steps below do not walk through the OAuth flow. Please update the steps to walk through OAuth steps a user will take to install the integration as mentioned above. 

Users should not be creating api or app keys within their account as OAuth replaces the need for this.

- Finalize Integration: Once satisfied, click "Create" to establish the Datadog connector. It will be deployed shortly.

    ![Deploying][5]

- View Host Resources: After successful deployment, you'll be able to see a list of host resources from your Datadog account within Onepane.

    ![Host Resource][6]

- Map Resources: Onepane allows you to map these Datadog resources to your existing Onepane resources for unified view of incidents and events across your infrastructure

    ![Onepane Mapping][7]

With these steps complete, you'll have successfully integrated Datadog with Onepane

For more details, refer to the [Onepane documentation][2]

## Uninstallation

Here's how to delete the Datadog integration from your Onepane console:

- Access Integrations: Navigate to the "Integrations" section within the Onepane console.

- Locate Datadog: Find the Datadog integration from the list.

- Initiate Deletion: Click the three dots menu (ellipsis) in the top right corner of the Datadog integration tile. Select "Delete" from the options.

- Confirm Deletion: A confirmation dialog might appear. If you're certain you want to proceed, click "Confirm" to remove the Datadog integration.

> By following these steps, you'll successfully disconnect your Datadog account from Onepane.

Reviewer: This should follow the oauth steps to uninstall the integration

# Onepane

## Overview

> Onepane is a GenAI tool that enhances incident resolution through automated Root Cause Analysis (RCA) by integrating seamlessly with monitoring tools like Datadog.

Reviewer 2: ```suggestion
Onepane is a GenAI tool that enhances incident resolution through automated root cause analysis (RCA) by integrating seamlessly with monitoring tools like Datadog.
```

By integrating Onepane with Datadog, customers gain rapid incident correlation with changes and infrastructure events, significantly reducing troubleshooting time and improving operational efficiency.

> The integration tracks incident data, infrastructure events, and change data (like code updates), providing valuable insights that help teams quickly identify root causes and prevent future issues.

Reviewer 2: ```suggestion
The integration tracks incident data, infrastructure events, and change data (such as code updates), providing valuable insights that help teams quickly identify root causes and prevent future issues.
```

## Setup

**In Datadog:**

> - Go to **Integrations**, select the **Onepane** tile, and click **Install Integration**.

Reviewer 2: ```suggestion
- Navigate to **Integrations**, select the **Onepane** tile, click **Install Integration**.
```

- Click **Connect Accounts** to begin authorizing the integration. You will be redirected to [console.onepane.ai][2].

**In Onepane:**

- Log in if you're not already logged in with **Onepane**.

> - Please review the  the permissions and click on the 'Connect with datadog' button in the prompt.

Reviewer 2: ```suggestion
- Review the permissions and click the **Connect with Datadog** button in the prompt.
```

    ![Onepane_prompt][10]

> - You will be redirected back into Datadog to complete the authorization

Reviewer 2: ```suggestion
- You will be redirected back into Datadog to complete the authorization.
```

- Click the **Authorize** button to complete the setup and be redirected back to the Onepane site.

- Provide a name for your Datadog connector.

- Click **Create** to complete the integration. The Datadog connector will be deployed shortly.

    ![Deploying][5]

- View Host Resources: After successful deployment, you'll be able to see a list of host resources from your Datadog account within Onepane.

    ![Host Resource][6]

> - Map Resources: Onepane allows you to map these Datadog resources to your existing Onepane resources for unified view of incidents and events across your infrastructure

Reviewer 2: ```suggestion
- Map Resources: Onepane allows you to map these Datadog resources to your existing Onepane resources for a unified view of incidents and events across your infrastructure.
```

    ![Onepane Mapping][7]

> With these steps complete, you'll have successfully integrated Datadog with Onepane

Reviewer 2: ```suggestion
With these steps complete, you'll have successfully integrated Datadog with Onepane.
```
